### PR TITLE
Refactor your code

### PR DESCRIPTION
**Pull request by anderCM**
At this point your codebase is pretty big. The `main.rb` file may be especially large at this point. Let's try to clean things up!

- Make sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Follow SOLID principles.
- Pay special attention to the "Single-responsibility Principle".
  - Make sure that the role of `main.rb` is only to start your code and ask the user for the initial choice of options (or to quit).
  - Extract some pieces of logic into separate class(es).

### CODE REVIEW

File: run.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      exit
+    else
+      create(option)
+    end
+  end
+
+  def create(option)
+    case option
+    when '3'
+      create_person(@peoples)
```
Comment (Gambit142) [inline run.rb]: - [x] Kindly refactor the `create_person`, `create_teacher`, `create_student`, `create_book`, and `create_rental` methods. In these methods, all codes responsible for **_taking input and validating inputs_** should be in a separate method, module or class while these methods should be responsible for creating the **_class instances_**. Kindly refactor the code. 👍 💻 

After the refactor, please check that your application functions properly. 👍 😎

File: run.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  def show(option)
+    case option
+    when '1'
+      list_all_books(@books)
+    when '2'
+      list_all_peoples(@peoples)
+    when '6'
+      list_rentals(@rentals, @peoples)
+    when '7'
```
Comment (Gambit142) [inline run.rb]: - [x] In the `list_rentals` method, kindly put all, codes responsible for **_taking the user's input and validating the user's inputs_** should be in a separate method, module or class while the list_rental method should be responsible for listing the rental instances as we have it in the `list_all_books` and `list_all_peoples` methods. Kindly implement this change. 👍 💻 

After the refactor, please check that your application functions properly. 👍 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @anderCM @Feliverse 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- App module is in a new file ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: run.rb (hunk 1 above)
Comment (Feliverse) [reply run.rb]: @Gambit142 ,  Ok, I will refactor to classes this methods and separate in modules.
Thank you

File: run.rb (hunk 2 above)
Comment (Feliverse) [reply run.rb]: @Gambit142  Nice observation, I will refactor these methods to validate inputs separately modules

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @anderCM & @Feliverse, 👋 👋  

Your project is complete! Great job for following the project requirement🥇 and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Descriptive Pull Request title and summary. ✔️
- Refactor methods using YAGNI, KISS, DRY, and Ruby style guide ✔️
- Pair Programming activity implemented ✔️
- SOLID principles Used to refactor classes ✔️
- Good use of Git flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**